workers/unity/Packages/com.improbable.gdk.tools/Plugins/PluginPostprocessor.cs

```diff
@@ -0,0 +1,135 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace Improbable.Gdk.Tools
+{
+    internal class PluginPostprocessor : AssetPostprocessor
+    {
+        private static readonly List<PluginDirectoryCompatibility> PluginsCompatibilityList = new List<PluginDirectoryCompatibility>
+        {
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.StandaloneOSX
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.StandaloneLinux64
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.StandaloneWindows64,
+                CPUType.X86_64
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.StandaloneWindows,
+                CPUType.X86
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.Android,
+                CPUType.Arm64
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.Android,
+                CPUType.ARMv7
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.Android,
+                CPUType.X86
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Core,
+                BuildTarget.iOS
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Sdk,
+                new List<BuildTarget> { BuildTarget.iOS }
+                ),
+            new PluginDirectoryCompatibility(
+                PluginType.Sdk,
+                BuildTarget.iOS,
+                CPUType.Unused,
+                false
+                ),
+        };
+
+        internal static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths)
+        {
+            if (importedAssets.Any(DownloadCoreSdk.IsImprobablePlugin))
+            {
+                SetPluginsCompatibility();
+            }
+        }
+
+        /// <summary>
+        ///     Sets plugin platform compatibility based on directory structure
+        /// </summary>
+        internal static void SetPluginsCompatibility()
+        {
+            AssetDatabase.StartAssetEditing();
+            foreach (var pluginDirectoryCompatibility in PluginsCompatibilityList)
+            {
+                if (!Directory.Exists(pluginDirectoryCompatibility.PluginPath))
+                {
+                    Debug.Log("directory doesn't exist " + pluginDirectoryCompatibility.PluginPath);
+                    continue;
+                }
+
+                var pluginPaths = AssetDatabase.FindAssets(string.Empty, new[] { pluginDirectoryCompatibility.PluginPath });
```

Comment:
driveby opportunity: `pluginPaths` is incorrect, it should be `pluginGuids`

Or, `pluginGuidsInFolder`?